src/tests/end2end/GruTests.cpp

```diff
@@ -201,5 +203,57 @@ TEST_F(GruTests, GruWithoutInitialHiddenState) {
 
+    TestGru(input, weight, recurrentWeight, steps, hiddenSize, {expected}, &options);
+}
+
+TEST_F(GruTests, GruWithReturnSequenceTrue) {
+    const int32_t steps = 2;
+    const int32_t batchSize = 3;
+    const int32_t inputSize = 3;
+    const int32_t hiddenSize = 5;
+    const int32_t numDirections = 1;
+
+    const std::vector<int32_t> inputShape = {steps, batchSize, inputSize};
+    const std::vector<float> inputData = {1,  2,  3,  4,  5,  6,  7,  8,  9,
+                                          10, 11, 12, 13, 14, 15, 16, 17, 18};
+    Tensor input = {inputShape, inputData};
+    const std::vector<int32_t> weightShape = {numDirections, 3 * hiddenSize, inputSize};
+    const std::vector<float> weightData(numDirections * 3 * hiddenSize * inputSize, 0.1);
+    Tensor weight = {weightShape, weightData};
+    const std::vector<int32_t> recurrentWeightShape = {numDirections, 3 * hiddenSize, hiddenSize};
+    const std::vector<float> recurrentWeightData(numDirections * 3 * hiddenSize * hiddenSize, 0.1);
+    Tensor recurrentWeight = {recurrentWeightShape, recurrentWeightData};
+    const std::vector<int32_t> biasShape = {numDirections, 3 * hiddenSize};
+    const std::vector<float> biasData(numDirections * 3 * hiddenSize, 0.1);
+    const ml::Operand bias =
+        utils::BuildConstant(builder, biasShape, biasData.data(), biasData.size() * sizeof(float));
+    const std::vector<int32_t> recurrentBiasShape = {numDirections, 3 * hiddenSize};
+    const std::vector<float> recurrentBiasData(numDirections * 3 * hiddenSize, 0);
+    const ml::Operand recurrentBias =
+        utils::BuildConstant(builder, recurrentBiasShape, recurrentBiasData.data(),
+                             recurrentBiasData.size() * sizeof(float));
+
+    ml::GruOptions options = {};
+    options.bias = bias;
+    options.recurrentBias = recurrentBias;
+    options.resetAfter = false;
+    options.returnSequence = true;
+
+    const std::vector<int32_t> expectedShape0 = {numDirections, batchSize, hiddenSize};
+    const std::vector<int32_t> expectedShape1 = {steps, numDirections, batchSize, hiddenSize};
+    const std::vector<float> expectedValue0 = {
+        0.22391089, 0.22391089, 0.22391089, 0.22391089, 0.22391089, 0.1653014, 0.1653014, 0.1653014,
+        0.1653014,  0.1653014,  0.0797327,  0.0797327,  0.0797327,  0.0797327, 0.0797327};
+    const std::vector<float> expectedValue1 = {
+        0.20053661, 0.20053661, 0.20053661, 0.20053661, 0.20053661, 0.15482338,
+        0.15482338, 0.15482338, 0.15482338, 0.15482338, 0.07484276, 0.07484276,
+        0.07484276, 0.07484276, 0.07484276, 0.22391089, 0.22391089, 0.22391089,
+        0.22391089, 0.22391089, 0.1653014,  0.1653014,  0.1653014,  0.1653014,
+        0.1653014,  0.0797327,  0.0797327,  0.0797327,  0.0797327,  0.0797327,
+    };
+    Tensor expected0 = {expectedShape0, expectedValue0};
+    Tensor expected1 = {expectedShape1, expectedValue1};
+    std::vector<Tensor> expected = {expected0, expected1};
```

Comment:
`std::move(expected)` in L.258 because the `TestGru` define the argument is `const std::vector<Tensor> expected`.
You can also reduce the local variable such as `expectedShape0, expectedValue0` with using the data directly.

Reply:
Good suggestion, optimized the code. PTAL, thanks!